A new task page for assigining a pod to nodes using node affinity (#19265)
kubernetes/website: content/en/docs/tasks/configure-pod-container/assign-pods-nodes-using-node-affinity.md

---
title: Assign Pods to Nodes using Node Affinity
content_template: templates/task
weight: 120
---

{{% capture overview %}}
This page shows how to assign a Kubernetes Pod to a particular node using Node Affinity in a
Kubernetes cluster.
{{% /capture %}}

{{% capture prerequisites %}}

> {{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}

Reviewer: When using `{{< version-check >}}`, make sure to set `min-kubernetes-server-version` in the front matter metadata.

Author: @sftim , I am not sure hot to add `min-kubernetes-server-version` Can you please point me to exisitng page where this is used. Or can I remove `{{< version-check >}} if it is not required.`

Reviewer: Have a look at line 3 of https://raw.githubusercontent.com/kubernetes/website/23a28d6dd288293d6d2970a9381f48753b7969fe/content/en/docs/tasks/configure-pod-container/share-process-namespace.md#L3

{{% /capture %}}

{{% capture steps %}}

## Add a label to a node

1. List the nodes in your cluster, along with their labels:

    ```shell
    kubectl get nodes --show-labels
    ```

    The output is similar to this:

    ```shell
    NAME      STATUS    ROLES    AGE     VERSION        LABELS
    worker0   Ready     <none>   1d      v1.13.0        ...,kubernetes.io/hostname=worker0
    worker1   Ready     <none>   1d      v1.13.0        ...,kubernetes.io/hostname=worker1
    worker2   Ready     <none>   1d      v1.13.0        ...,kubernetes.io/hostname=worker2
    ```
1. Chose one of your nodes, and add a label to it:

    ```shell
    kubectl label nodes <your-node-name> disktype=ssd
    ```

    where `<your-node-name>` is the name of your chosen node.

1. Verify that your chosen node has a `disktype=ssd` label:

    ```shell
    kubectl get nodes --show-labels
    ```

    The output is similar to this:

>     ```shell

Reviewer: This sample output is not a valid shell program.

    NAME      STATUS    ROLES    AGE     VERSION        LABELS
    worker0   Ready     <none>   1d      v1.13.0        ...,disktype=ssd,kubernetes.io/hostname=worker0
    worker1   Ready     <none>   1d      v1.13.0        ...,kubernetes.io/hostname=worker1
    worker2   Ready     <none>   1d      v1.13.0        ...,kubernetes.io/hostname=worker2
    ```

    In the preceding output, you can see that the `worker0` node has a
    `disktype=ssd` label.

> ## Assign pod to node using required node affinity

Reviewer: ```suggestion
## Create a Pod with scheduling constraints
```
?
I'm not sure about wording but I think it's good to mention scheduling.

> This pod configuration file describes a pod that has a `requiredDuringSchedulingIgnoredDuringExecution` node affinity,`disktype: ssd`. 

Reviewer: ```suggestion
This manifest describes a Pod that has a `requiredDuringSchedulingIgnoredDuringExecution` node affinity,`disktype: ssd`.
```

This means that the pod will get scheduled only on a node that has a `disktype=ssd` label. 

{{< codenew file="pods/pod-nginx-required-affinity.yaml" >}}

> 1. Use the configuration file to create a pod that will get scheduled on your

Reviewer: ```suggestion
1. Apply the manifest to create a pod that is scheduled onto your
```

   chosen node:
    
    ```shell
    kubectl apply -f https://k8s.io/examples/pods/pod-nginx-required-affinity.yaml
    ```

1. Verify that the pod is running on your chosen node:

    ```shell
    kubectl get pods --output=wide
    ```

    The output is similar to this:
    

>     ```shell

Reviewer: This is sample output, not shell code.

    NAME     READY     STATUS    RESTARTS   AGE    IP           NODE
    nginx    1/1       Running   0          13s    10.200.0.4   worker0
    ```
## Assign pod to node using preferred node affinity

This pod configuration file describes a pod that has a `preferredDuringSchedulingIgnoredDuringExecution` node affinity,`disktype: ssd`. 
This means that the pod will prefer a node that has a `disktype=ssd` label. 

{{< codenew file="pods/pod-nginx-preferred-affinity.yaml" >}}

1. Use the configuration file to create a pod that will get scheduled on your
   chosen node:
    
    ```shell
    kubectl apply -f https://k8s.io/examples/pods/pod-nginx-preferred-affinity.yaml
    ```

1. Verify that the pod is running on your chosen node:

    ```shell
    kubectl get pods --output=wide
    ```

    The output is similar to this:
    

>     ```shell

Reviewer: This is sample output, not shell code
